cli/tests/test_cmd_upgrade.py

```
@@ -5117,72 +5153,111 @@ def test_incomplete_hard_cut_rollback_never_claims_rolled_back(self):
+            "provenance": {"binary_version": "0.8.8"},
+        }
         with (
             patch("defenseclaw.gateway.OrchestratorClient", return_value=client),
-            patch("defenseclaw.commands.cmd_upgrade.ux.ok") as ok,
             patch(
                 "defenseclaw.commands.cmd_upgrade.time.monotonic",
-                side_effect=[0.0, 0.0, 0.0, 0.0],
+                side_effect=itertools.count(0.0, 2.0).__next__,
             ),
             patch("defenseclaw.commands.cmd_upgrade.time.sleep"),
+            self.assertRaises(SystemExit),
```

Ruff (0.15.21)

[warning] Use `pytest.raises` instead of unittest-style `assertRaises`

Replace `assertRaises` with `pytest.raises`

(PT027)